pkg/batch/batch.go

```diff
@@ -0,0 +1,77 @@
+package batch
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/loopfz/gadgeto/zesty"
+
+	"github.com/ovh/utask/models/task"
+	"github.com/ovh/utask/models/tasktemplate"
+	"github.com/ovh/utask/pkg/taskutils"
+)
+
+// TaskArgs holds arguments needed to create tasks in a batch
+type TaskArgs struct {
+	TemplateName     string                   // Mandatory
+	Inputs           []map[string]interface{} // Mandatory
+	CommonInput      map[string]interface{}   // Optional
+	Comment          string                   // Optional
+	WatcherUsernames []string                 // Optional
+	WatcherGroups    []string                 // Optional
+	Tags             map[string]string        // Optional
+}
+
+// Populate creates and adds new tasks to a given batch.
+// All tasks share a common batchID which can be used as a listing filter.
+// The [constants.SubtaskTagParentTaskID] tag can be set in the Tags to link the newly created tasks to another
+// existing task, making it the parent of the batch. A parent task is resumed everytime a child task finishes.
+func Populate(ctx context.Context, batch *task.Batch, dbp zesty.DBProvider, args TaskArgs) ([]string, error) {
+	tt, err := tasktemplate.LoadFromName(dbp, args.TemplateName)
+	if err != nil {
+		return []string{}, err
+	}
+
+	taskIDs := make([]string, 0, len(args.Inputs))
+	for _, inp := range args.Inputs {
+		input, err := conjMap(args.CommonInput, inp)
+		if err != nil {
+			return []string{}, err
+		}
+
+		t, err := taskutils.CreateTask(
+			ctx,
+			dbp,
+			tt,
+			args.WatcherUsernames,
+			args.WatcherGroups,
+			[]string{},
+			[]string{},
+			input,
+			batch,
+			args.Comment,
+			nil,
+			args.Tags,
+		)
+		if err != nil {
+			return []string{}, err
+		}
+		taskIDs = append(taskIDs, t.PublicID)
+	}
+	return taskIDs, nil
+}
+
+func conjMap(common, particular map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
The name of this function does not clearly reflect its usage.
It does a merge/union of two map while checking that common keys are not overriden => btw, I'm skeptical of the check line 71/72. It seems legitimate to want to define a "default/common" value for all tasks, while still overriding it for some tasks, no?

Reply:
This function used to sit elsewhere (https://github.com/ovh/utask/blob/master/api/handler/batch.go#L80), I simply moved it and kept the exact same behaviour. Changing its name (something like `mergeMaps` seems adequate) won't have any side effect, but I'm not sure changing its behaviour is 100% safe.
That being said, allowing each task to override the default value is a good idea and seems like a valid use case.